starter_code/src/movies.js

```diff
@@ -1,18 +1,110 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+function horasF(str){
+  let horas = 0
+  let minutos = 0
+  let total = 0
+  let h = str.split(" ")
+  horas = parseInt(h)
+  console.log(h)
+  //si el array hora existe y no hay minutos retorna las horas en minutos
+  //minutos
 
+  if(!h[1]&&h[0].includes("h")){
+    total = horas * 60
+  } else if(!h[1]){
+    total=parseInt(h[0])
+  }
 
-// Get the average of all rates with 2 decimals 
+  //si hay horas y minutos
+  else {
+    let j = h[1].split("min")
+    minutos = parseInt(j[0])
+    //console.log("minutos: "+minutos)
+    //console.log("horas: "+ horas)
+    total = (horas*60)+minutos
+   // console.log(typeOf(total))
+  }
+    return (total)
+}
 
 
-// Get the average of Drama Movies
+function turnHoursToMinutes (movies){
+  let movmap=  movies.map((e)=>{
+   e.duration=horasF(e.duration)
+   return e
+  })
 
+  console.log(movmap)
+  return movmap
+}
 
-// Order by time duration, in growing order
+function ratesAverage(arr) {
+  let rate = arr.map((e) => {
+    return e.rate
+  })
+  let prom = rate.reduce((cont, actValue) => {
+    return cont + Number(actValue)
+  }, 0)
 
+  return parseFloat((prom / arr.length).toFixed(2))
+}
 
-// How many movies did STEVEN SPIELBERG
+ratesAverage(movies)
 
+const dramaMoviesRate = arr => {
+  
+
+
+  let result = arr.filter(e => e.genre.includes('Drama'));
+
+  let rateD =result.map((e) => {
+    return e.rate
+  })
+  
+  let prom = rateD.reduce((cont, actual) => {
+    return cont + Number(actual)
+  }, 0)
+  
+   //console.log(parseFloat((prom / arr.length).toFixed(2)))
+   return (parseFloat((prom / rateD.length).toFixed(2)))
+  }
+  
+  dramaMoviesRate(movies)
+
+  function orderByDuration(arr){
+    arr.duration
+    let newArray=[]
+    newArray= arr.sort((a, b) => a.duration.localeCompare(b.duration))
+    return newArray.reverse()
+  
+  }
+
+  function howManyMovies(arr){
+    let result = arr.filter(e => e.genre.includes('Drama'))
+    .filter(e => e.director.includes('Steven Spielberg'))
+
+    if(result.length == 0) {
+      return undefined
+    }
+    return "Steven Spielberg directed 4 drama movies!"
+  }
+  
+  howManyMovies(movies)
+
+
+  const orderAlphabetically = arr => {
+    let aux = []
+    let test = arr.sort((a, b) => {
+      return a.title.localeCompare(b.title)
+    })
+    for(i = 0; i < 20; i++) {
+      aux.push(test[i].title)
```

Comment:
Cuidado: para arreglos con menos de 20 elementos, esto va a generar un error. Aquí la idea sería hacer un sort del arreglo (cómo hicieron ustedes) y luego nada más hacerle un `.slice`. El for loop solo funcionaría si le agregaran una condición para que no siguiera hasta 20 cuando el arreglo tiene menos de 20 elementos.